src/main/java/com/cloudbees/jenkins/support/configfiles/SecretHandler.java

```diff
@@ -78,7 +77,7 @@ public void characters(char[] ch, int start, int length) throws SAXException {
             }
         };
         Source src = new SAXSource(xr, new InputSource(new StringReader(FileUtils.readFileToString(xmlFile))));
-        File patchedFile = File.createTempFile("patched", ".xml", SupportPlugin.getRootDirectory());
+        File patchedFile = File.createTempFile("patched", ".xml");
```

Comment:
I doubt many setups configure the java tmp directory to be private, despite what you are calling out in the JIRA.
These files are created with the default umask - and although we attempt to patch them we may not - so it may leak secrets world readable.
so a :bug: on at least not restricting read access here.